client/landing/gutenboarding/stepper-onboarding/hooks/use-step-navigation.tsx

```diff
@@ -0,0 +1,24 @@
+import { useCallback } from '@wordpress/element';
+import { useHistory } from 'react-router-dom';
+import { useStepRouteParam } from '../path';
+import { useSteps } from './use-steps';
+
+export function useStepNavigation() {
+	const step = useStepRouteParam();
+	const steps = useSteps();
+	const history = useHistory();
+
+	const goNext = useCallback( () => {
+		const index = steps.indexOf( step );
+		const goTo = steps[ Math.min( index + 1, steps.length - 1 ) ];
+		history.push( `/${ goTo }` );
+	}, [ step, steps, history ] );
+
+	const goBack = useCallback( () => {
```

Comment:
In a similar thought to what @p-jackson mentioned above, what if we combined both of these into one function that accepts a parameter to go ⇠, ⇢, or to a specific index? Just a thought, I don't see any problems with what you have already and the only real advantage I see to switching this would be the option to jump to an index which could allow for a "Start over" button. Something like this?

```javascript
	const goToStep = useCallback(
		( newStep ) => {
			const index = steps.indexOf( step );
			const goTo = () => {
				switch ( newStep ) {
					case 'previous':
						return steps[ Math.max( index - 1, 0 ) ];
					case 'next':
						return steps[ Math.min( index + 1, steps.length - 1 ) ];
					default:
						return newStep;
				}
			};
			history.push( `/${ goTo() }` );
		},
		[ step, steps, history ]
	);
```

Reply:
This totally works but needlessly makes it more complicated.